src/bin/command_prelude.rs

```diff
@@ -196,9 +198,28 @@ pub trait ArgMatchesExt {
         Ok(arg)
     }
 
+    /// Returns value of the `name` command-line argument as an absolute path
+    fn value_of_path(&self, name: &str, config: &Config) -> Option<PathBuf> {
```

Comment:
Could this be called `normalized_value_of_path` for clarity? I know in rustc we'd want to avoid a method like this at all costs due to presenting the paths later on, but we don't really do that in Cargo (or do it in a limited capacity) so the normalization we do here is fine for Cargo but just not applicable to all situations.

Reply:
An excellent observation! Actually, I'd say that we mustn't normalize paths there, except for for a single case, which could be moved inside `root_manifest`. And that is needed because https://github.com/rust-lang/cargo/issues/3586. 

Updated the PR!